pkg/api/defaults.go

```diff
@@ -306,6 +306,12 @@ func (cs *ContainerService) setOrchestratorDefaults(isUpgrade, isScale bool) {
 			}
 		}
 
+		if strings.ToLower(a.OrchestratorProfile.KubernetesConfig.LoadBalancerSku) == strings.ToLower(BasicLoadBalancerSku) {
```

Comment:
Here we are doing some "convenient" string rationalization so that by the time we compose our Azure API spec, we are always using LB SKU values of the "capitalized" form.